docs/architecture/adr-069-flexible-node-intitalization.md

```diff
@@ -0,0 +1,164 @@
+
+- The implementation of the `node` service, doesn't really need the
+  ability to introspect the implementations of reactors in most
+  cases.
+
+- The exception to this, is the cluster of
+  consensus/statesync/blockhain (which depend on eachother,) because
+  `node` itself contains some of their specific initialization logic.
+
+- While the RPC environment construction happens in `node`, this can
+  happen early during `node` construction rather than at
+  initialization time, which reduces the need for the implementation
+  of the node object to track this state.
+
+Ideally, it seems like a node instance should be expressable simply as
+a list of `service.Services` and any knowledge of the internals of any
+of these services could be limited to their constructors. There is a
+sketch of a generic and simple `groupService` implementation, to handle a
+desecrate stage or sequence of services.  As some services have
+dependencies during construction or initialization and as a starting
+point, a cluster of dependencies can be wrapped as a single
+`service.Service`.
+
+## Goals
+
+- Provide a more flexible internal framework for initializing tendermint
+  nodes to make the initatilization process less hard-coded by the
+  implementation of the node objects. 
+  
+  - Reactors should not need to expose their interfaces *within* the
+    implementation of the node type, except in the context of some groups of
+    service. 
+
+- Expose some "unsafe"  way of replacing (some?) existing components, and or
+  inserting arbitrary reactors or services into the tendermint node's
+  initialization.
+  
+  - This may include creating an exported, top-level "unsafe" package with
+    several interfaces and constructors, or simply a function in the
+    node package. 
+
+- Refactor and simplify the process start/initialization logic,
+  separately from--but as a precursor to--permitting user replacement
+  of components.
+
+## Non-Goals
+
+- Fully abstract dependency-based process initialization. 
+
+- Fully abstract replacement of core tendermint components without vendoring.
+
+- Parallelized initialization of components.
+
+## Questions
+
+- To what extent does this new initialization framework need to accommodate
+  the legacy p2p stack? Would it be possible to delay a great deal of this
+  work to the 0.36 cycle to avoid this complexity? 
+  
+  - Answer: _depends on timing_, and the requirement to ship pluggable
+    reactors in 0.35. 
+  
+- There's clear and common interest in injecting some kind of mempool, and
+  intuitively this makes sense as the highest priority.
+  
+  - Does the existing `mempool.Mempool` interface make sense to expose and
+    commit to, or should we make a smaller interface? 
+    
+	- Answer: _yes_, smaller mempool interface is better. 
+
+  - Is it reasonable for users to bring their own mempool without bringing
+    their own reactor?
+	
+	- Answer: _probably not_, but this might be a function of how the
```

Comment:
When this ADR gets updated, it would be nice to include some of the thoughts from the above comment thread in this "answer" section